--- commoncpp2-1.7.1/src/cidr.cpp.orig	2009-10-23 23:38:56.094302646 +0200
+++ commoncpp2-1.7.1/src/cidr.cpp	2009-10-23 23:41:40.582997710 +0200
@@ -202,7 +202,7 @@
 #if defined(_MSC_VER) && _MSC_VER >= 1500
 	ep = (char *)strchr(cp, '/');
 #else
- 	ep = strchr(cp, '/');
+ 	ep = (char *)strchr(cp, '/');
 #endif
 
 	if(ep)
@@ -332,7 +332,7 @@
 	memset(&netmask, 0, sizeof(netmask));
 	bitset((bit_t *)&netmask, getMask(cp));
 	setString(cbuf, sizeof(cbuf), cp);
-	ep = strchr(cp, '/');
+ 	ep = (char *)strchr(cp, '/');
 	if(ep)
 		*ep = 0;
 
